### PR TITLE
[WIP] Refactor

### PR DESCRIPTION
**Pull request by fmrico**
Hi devs,

This PR contains a **huge** refactor. The legacy code was terrible and was not linted. It broke some of the architectonical design. For example, it publishes TFs directly and many other errors, like the QoS. It had a lot of unnecessary parameters. The structure was incorrect. Besides, I have changed the package name to make it uniform with the other MOCAP4ROS2 packages.

This PR shouldn't be merged until it is tested in the real system in the lab next week. That's why it is a WIP (Work In Progress).

I also would like to hear from @aarsht7 and @jmguerreroh their suggestions in this refactoring.

Best
Francisco

### CODE REVIEW

File: mocap_optitrack_driver/NatNetSDK/.gitignore [hunk 1]
```diff
@@ -0,0 +1,3 @@
+include
+lib
+samples
```
Comment (aarsht7) [inline mocap_optitrack_driver/NatNetSDK/.gitignore:3]: Should we keep only one gitignore or each in individual folder? 

If you want only one gitignore, 
[This](https://github.com/MOCAP4ROS2-Project/mocap4ros2_optitrack/blob/main/.gitignore) gitignore will do the same functionality as this one. Just change the name of main folder as it has been changed globally in the repo.

File: mocap_optitrack_driver/cmake/FindNATNETSDK_CPP.cmake [hunk 2]
```diff
@@ -0,0 +1,46 @@
...
+
+find_library(
+    LIBNATNET_SDK_LIBRARY
+    NAMES NatNet)
+
+find_path(
+  LIBNATNET_SDK_INCLUDE_DIR
+  NAMES NatNetClient.h NatNetCAPI.h NatNetTypes.h
+  PATHS /usr/local/include/NatNetSDK)
+
```
Comment (aarsht7) [inline mocap_optitrack_driver/cmake/FindNATNETSDK_CPP.cmake:30]: I have modified the cmakelist to include all the header files from natnet sdk. If it looks good to you, we can skip this cmake file as it is no more called from current cmakelist.

File: mocap_optitrack_driver/NatNetSDK/.gitignore (hunk 1 above)
Comment (fmrico) [reply mocap_optitrack_driver/NatNetSDK/.gitignore:3]: Ok, lets adjust in in next Prs

File: mocap_optitrack_driver/cmake/FindNATNETSDK_CPP.cmake (hunk 2 above)
Comment (fmrico) [reply mocap_optitrack_driver/cmake/FindNATNETSDK_CPP.cmake:30]: It's up to you ;). Adjust in a separate PR, please.

**Comment by fmrico** [comment]
Hi all,

The driver is fully working and tested in the real system.

Merging...

**Pull request merged**